tensorflow_io/arrow/kernels/arrow_kernels.cc

```diff
@@ -0,0 +1,266 @@
+    OP_REQUIRES(context, !memcmp(buffer.data(), kFeatherMagicBytes, header_length), errors::InvalidArgument("not a feather file"));
+
+    OP_REQUIRES_OK(context, file->Read(size - footer_length, footer_length, &result, &buffer[0]));
+    OP_REQUIRES(context, !memcmp(buffer.data() + sizeof(uint32), kFeatherMagicBytes, footer_length - sizeof(uint32)), errors::InvalidArgument("incomplete feather file"));
+
+    uint32 metadata_length = *reinterpret_cast<const uint32*>(buffer.data());
+
+    buffer.resize(metadata_length);
+
+    OP_REQUIRES_OK(context, file->Read(size - footer_length - metadata_length, metadata_length, &result, &buffer[0]));
+
+    const ::arrow::ipc::feather::fbs::CTable* table = ::arrow::ipc::feather::fbs::GetCTable(buffer.data());
+
+    OP_REQUIRES(context, (table->version() >= ::arrow::ipc::feather::kFeatherVersion), errors::InvalidArgument("feather file is old: ", table->version(), " vs. ", ::arrow::ipc::feather::kFeatherVersion));
+
+    std::vector<string> columns;
+    std::vector<string> dtypes;
+    std::vector<int64> counts;
+    columns.reserve(table->columns()->size());
+    dtypes.reserve(table->columns()->size());
+    counts.reserve(table->columns()->size());
+
+    for (int64 i = 0; i < table->columns()->size(); i++) {
+      string dtype = "";
+      switch (table->columns()->Get(i)->values()->type()) {
+      case ::arrow::ipc::feather::fbs::Type_BOOL:
+        dtype = "bool";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_INT8:
+        dtype = "int8";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_INT16:
+        dtype = "int16";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_INT32:
+        dtype = "int32";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_INT64:
+        dtype = "int64";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_UINT8:
+        dtype = "uint8";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_UINT16:
+        dtype = "uint16";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_UINT32:
+        dtype = "uint32";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_UINT64:
+        dtype = "uint64";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_FLOAT:
+        dtype = "float";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_DOUBLE:
+        dtype = "double";
+        break;
+      case ::arrow::ipc::feather::fbs::Type_UTF8:
+      case ::arrow::ipc::feather::fbs::Type_BINARY:
+      case ::arrow::ipc::feather::fbs::Type_CATEGORY:
+      case ::arrow::ipc::feather::fbs::Type_TIMESTAMP:
+      case ::arrow::ipc::feather::fbs::Type_DATE:
+      case ::arrow::ipc::feather::fbs::Type_TIME:
+      // case ::arrow::ipc::feather::fbs::Type_LARGE_UTF8:
+      // case ::arrow::ipc::feather::fbs::Type_LARGE_BINARY:
+      default:
+        break;
+      }
+      if (dtype == "") {
+        continue;
```

Comment:
Thanks @BryanCutler. Updated the PR to change the dtype to `INVALID` and populate to the python process (so that it is possible to process at the higher level).